logback-core/src/main/java/ch/qos/logback/core/net/ssl/SSLConfigurableSocket.java

```diff
@@ -60,4 +64,28 @@ public void setWantClientAuth(boolean state) {
         delegate.setWantClientAuth(state);
     }
 
+    public void setDisableHostnameVerification(boolean disableHostnameVerification) {
+        if (disableHostnameVerification) {
+            return;
+        }
+        SSLParameters sslParameters = delegate.getSSLParameters();
+        try {
+            /*
+             * SSLParameters.setEndpointIdentificationAlgorithm(...)
+             * is only available from Java 7, so we use reflection
+             * to check whether it is available.
+             */
+            Method setEndpointIdentificationAlgorithmMethod = SSLParameters.class.getMethod("setEndpointIdentificationAlgorithm", String.class);
+            setEndpointIdentificationAlgorithmMethod.invoke(sslParameters, "HTTPS");
+            delegate.setSSLParameters(sslParameters);
+        } catch (NoSuchMethodException e) {
+            // We don't do anything when the method is not available.
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        } catch (IllegalArgumentException e) {
+            throw new RuntimeException(e);
```

Comment:
I don't think you've looked close enough :smile: Like FindBugs, Codacy allows configuration of checks:

![screen shot 2016-04-07 at 12 02 03 pm](https://cloud.githubusercontent.com/assets/26580/14359791/b976b19a-fcb8-11e5-8be6-9960f5335618.png)

I'm not sure if you were alluding to this issue here as noise, but I think Codacy made a good point.